src/v/cloud_storage/manifest.h

```diff
@@ -170,12 +176,15 @@ class manifest final : public base_manifest {
 
     /// Check if the manifest contains particular segment
     bool contains(const key& key) const;
+    bool contains(const segment_name& key) const;
```

Comment:
this method was changed to use `key` in the previous commit and added back here, I wonder if there was some merge conflict

Reply:
I just added second overload.